conformance/tests/mesh-consumer-route.go

```diff
@@ -33,9 +33,7 @@ var MeshConsumerRoute = suite.ConformanceTest{
 	ShortName:   "MeshConsumerRoute",
 	Description: "An HTTPRoute in a namespace other than its parentRef's namespace only affects requests from the route's namespace",
 	Features: []features.SupportedFeature{
-		features.SupportMesh,
-		features.SupportHTTPRoute,
-		features.SupportHTTPRouteResponseHeaderModification,
```

Comment:
These cannot be part of the feature set. IMO this is due to a limitation of the framework. What we want is

```
ProvesFeatures: [features.SupportMeshConsumerRoute]
DependsOnFeatures: [features.SupportHTTPRoute, features.SupportHTTPRouteResponseHeaderModification]
```

That is, you cannot be SupportMeshConsumerRoute without this test, but you can be SupportHTTPRoute without this test. However, you need SupportHTTPRoute to be SupportMeshConsumerRoute due to testing implications

Reply:
IIUC, without this an implementation cannot claim `SupportHTTPRouteResponseHeaderModification` without passing this test.

This test does not **test** header modification. It **depends** on header modification as a mechanism to make assertions, but doesn't prove nor deny support for response header modifications.

Reply:
IMO this would be testing the intersection between two extended features, and no implementation should be required to pass tests that cover features they're not claiming to support (x-ref https://github.com/kubernetes-sigs/gateway-api/pull/3025#discussion_r1583399034). I realize that likely requires a docs update, but would rather do that than unnecessarily remove features from the list here.

Reply:
Sorry, but that really doesn't make sense to me. Let me draw an analogy outside of these tests...

Consider two features, SupportsJSON and SupportsObscureFeature.

In order to test `ObscureFeature`, we happen to send JSON requests. But JSON and ObscureFeature are not correlated in any way other than implementation details of the test, which happen to use JSON for convience.

IIUC, you are saying "An implementation cannot claim JSON support without supporting ObscureFeature". Is that accurate?

(JSON=RouteResponseHeaderModification,ObscureFeature=MeshConsumerRoute)

Reply:
I think we're agreeing with each other and I'm just not communicating well. I'm saying that any test that covers the intersection of two extended features should only be required if an implementation claims support for both extended features. By extension, that also means that we should always try to have at least some isolated test coverage for every extended feature that does not rely on other extended features.

Reply:
I think `features` already has the functionality you want? We only run a test if the implementation is claiming support for all features associated with the test. I don't really think we need any distinction here since the test is providing coverage for all features listed, the biggest thing we need to work on is ensuring that we actually have isolated test coverage for every extended feature that doesn't rely on other extended features. Open to suggestions for how to improve this further though.